eth_account/_utils/structured_data/validation.py

```diff
@@ -5,8 +5,8 @@
 )
 
 # Regexes
-IDENTIFIER_REGEX = r"^[a-zA-Z_$][a-zA-Z_$0-9]*$"
-TYPE_REGEX = r"^[a-zA-Z_$][a-zA-Z_$0-9]*(\[([1-9]\d*\b)*\])*$"
+IDENTIFIER_REGEX = r"^[a-zA-Z_$ ][a-zA-Z_$0-9 ]*$"
```

Comment:
Not sure if we should consider other [valid JSON property characters](https://stackoverflow.com/questions/8676011/which-characters-are-valid-invalid-in-a-json-key-name#:~:text=For%20Javascript%2C%20the%20characters%20requiring,objects%20with%20no%20escaping%20required.) as well? Might include tabs, escaped quotes, dashes, etc?

Reply:
This PR is in response to a particular issue #231 . Were `encode_structured_data`  to be sticking around, I would agree with all making all valid json characters valid. As it is, I'm working on a replacement `encode_typed_data` that will use different validation machinery entirely, and `encode_structured_data` will be de-emphasized and eventually deprecated. Since it will go away, I'm inclined to just leave it at spaces.